tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/metrics/AllMetricsTest.java

```diff
@@ -128,4 +140,19 @@ public void testAllMetrics() throws InterruptedException, ExecutionException {
         assertThat("bulkhead queue wait time histogram present", m.getBulkheadWaitingDuration().isPresent(), is(true));
     }
     
+    @Test
+    public void testMetricUnits() throws InterruptedException, ExecutionException {
+        // Call the method to ensure that all metrics get registered
+        allMetricsBean.doWork().get();
```

Comment:
If metrics are registered lazily, this isn't enough to trigger registering all metrics.

I'm not sure if metrics must be registered eagerly, I'll think about that a bit more.

Reply:
Actually, perhaps my implementation is a bit _too_ lazy -- if no retry ever happens, it doesn't register the `ft.retry.retries.total` metric, even though `ft.retry.calls.total` is registered. That's certainly unexpected. I'll treat it as a bug on my side.

Reply:
OK, good. Thanks for the test, found a few bugs :-)